diff -Naur linux-3.2.9.org/arch/arm/mach-versatile/core.c linux-3.2.9/arch/arm/mach-versatile/core.c
--- linux-3.2.9.org/arch/arm/mach-versatile/core.c	2012-03-01 01:32:49.000000000 +0100
+++ linux-3.2.9/arch/arm/mach-versatile/core.c	2012-05-24 19:35:19.264358053 +0200
@@ -642,24 +642,24 @@
 AMBA_DEVICE(ssp0,  "dev:f4",  SSP,      &ssp0_plat_data);
 
 static struct amba_device *amba_devs[] __initdata = {
-	&dmac_device,
+	/*&dmac_device,*/
 	&uart0_device,
 	&uart1_device,
 	&uart2_device,
-	&smc_device,
+	/*&smc_device,
 	&mpmc_device,
 	&clcd_device,
 	&sctl_device,
 	&wdog_device,
 	&gpio0_device,
-	&gpio1_device,
+	&gpio1_device,*/
 	&rtc_device,
-	&sci0_device,
+	/*&sci0_device,
 	&ssp0_device,
 	&aaci_device,
 	&mmc0_device,
 	&kmi0_device,
-	&kmi1_device,
+	&kmi1_device,*/
 };
 
 #ifdef CONFIG_OF
diff -Naur linux-3.2.9.org/arch/arm/mach-versatile/versatile_pb.c linux-3.2.9/arch/arm/mach-versatile/versatile_pb.c
--- linux-3.2.9.org/arch/arm/mach-versatile/versatile_pb.c	2012-03-01 01:32:49.000000000 +0100
+++ linux-3.2.9/arch/arm/mach-versatile/versatile_pb.c	2012-05-24 19:35:46.980493616 +0200
@@ -83,10 +83,10 @@
 
 static struct amba_device *amba_devs[] __initdata = {
 	&uart3_device,
-	&gpio2_device,
+	/*&gpio2_device,
 	&gpio3_device,
 	&sci1_device,
-	&mmc1_device,
+	&mmc1_device,*/
 };
 
 static void __init versatile_pb_init(void)
